data/data.go

```diff
@@ -0,0 +1,422 @@
+
+import (
+	"database/sql"
+	"errors"
+	"flag"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	driver  = flag.String("driver", "postgres", "Der benutzte sql-Treiber")
+	connect = flag.String("connect", "dbname=nnev user=anon host=/var/run/postgresql sslmode=disable", "Die Verbindusgsspezifikation")
+)
+
+// OpenDB opens a connection to the database with parameters derived from flags.
+func OpenDB() (*sql.DB, error) {
+	return sql.Open(*driver, *connect)
+}
+
+// Querier is an interface used to query the database. Both *sql.DB and *sql.Tx
+// implement it.
+type Querier interface {
+	// Query executes a query that returns rows, typically a SELECT. The args
+	// are for any placeholder parameters in the query.
+	Query(query string, args ...interface{}) (*sql.Rows, error)
+
+	// QueryRow executes a query that is expected to return at most one row.
+	// QueryRow always returns a non-nil value. Errors are deferred until Row's
+	// Scan method is called.
+	QueryRow(query string, args ...interface{}) *sql.Row
+}
+
+// Execer is an interface used to write to the database. Both *sql.DB and
+// *sql.Tx implement it.
+type Execer interface {
+	// Exec executes a query without returning any rows. The args are for any
+	// placeholder parameters in the query.
+	Exec(query string, args ...interface{}) (sql.Result, error)
+}
+
+// NullTime represents a time.Time that may be null. NullTime implements the
+// sql.scanner interface so it can be used as a scan destination, similar to
+// sql.NullString.
+type NullTime struct {
+	Time  time.Time
+	Valid bool // Vaid is true if Time is not NULL
+}
+
+// Scan implements the sql.scanner interface.
+func (n *NullTime) Scan(value interface{}) error {
+	if value == nil {
+		*n = NullTime{}
+		return nil
+	}
+	n.Valid = true
+	if t, ok := value.(time.Time); !ok {
+		return fmt.Errorf("can't save %T as time.Time")
+	} else {
+		n.Time = t
+	}
+	return nil
+}
+
+// scanner is an interface to retrieve values from an sql row. Both *sql.Row
+// and *sql.Rows implement it.
+type scanner interface {
+	// Scan copies the columns from the matched row into the values pointed at
+	// by dest. See the documentation on Rows.Scan for details. If more than
+	// one row matches the query, Scan uses the first row and discards the
+	// rest. If no row matches the query, Scan returns ErrNoRows.
+	Scan(dest ...interface{}) error
+}
+
+// row is an abstraction over the data elements to share code.
+type row interface {
+	// selectFragment returns an SQL query fragment of the form "SELECT field1,
+	// field2,... " with all columns for this type.
+	selectFragment() string
+
+	// scanFrom calls the scanners Scan method to retrieve the fields in the
+	// order output by selectFragment.
+	scanFrom(scanner) error
+}
+
+// queryRow is a convenience function to get a single data row.
+func queryRow(q Querier, r row, constrict string, args ...interface{}) error {
+	row := q.QueryRow(r.selectFragment()+constrict, args...)
```

Comment:
I find having a variable named the same way as a type name slightly confusing, although technically valid.